### ISSUE TITLE
Try Scikitlearn Clustering (k unknown)

### ISSUE THREAD

**Issue by sublime09** [sublime09/CrySmile#8]
Try MeanShift or VGBMM clustering algs.  
Based off of http://scikit-learn.org/stable/tutorial/machine_learning_map/index.html

**Comment by sublime09** [comment]
Emotion Labels come from issue #5

**Comment by sublime09** [comment]
Tried MeanShift, and can get about 5 clusters in ef_coffee when bandwidth estimate is .99.  
Next up: VGBMM

**Comment by sublime09** [comment]
From sklearn: 
The VBGMM object implements a variant of the Gaussian mixture model with variational inference algorithms. The API is identical to GMM. It is essentially a middle-ground between GMM and DPGMM, as it has some of the properties of the Dirichlet process.
Pros:
	Regularization:	due to the incorporation of prior information, variational solutions have less pathological special cases than expectation-maximization solutions. One can then use full covariance matrices in high dimensions or in cases where some components might be centered around a single point without risking divergence.
Cons:
	Bias: to regularize a model one has to add biases. The variational algorithm will bias all the means towards the origin (part of the prior information adds a “ghost point” in the origin to every mixture component) and it will bias the covariances to be more spherical. It will also, depending on the concentration parameter, bias the cluster structure either towards uniformity or towards a rich-get-richer scenario.
	Hyperparameters: this algorithm needs an extra hyperparameter that might need experimental tuning via cross-validation.

**Comment by sublime09** [comment]
Core problem: large n causes extreme slowdown in the scikiLearn clustering algorithms that we try to use.  Only the tiny coffee EmojiFrame (n=2000+) was finished.  Possible this issue is caused by slow pure-python instead of c++ implementations of scipy; see here: http://scikit-learn.org/stable/developers/performance.html
